Scripts/Run-SS-On-Exp/scripts/MicroColonyAnalysis.py

```diff
@@ -0,0 +1,322 @@
+    bacteria_orientation_in_current_time_step = [np.arctan2(cs[it].dir[1], cs[it].dir[0]) for it in cs if cs]
+
+    # center position
+    bacteria_x_center_in_current_timestep = [cs[it].pos[0] for it in cs]
+    bacteria_y_center_in_current_timestep = [cs[it].pos[1] for it in cs]
+
+    # cell age
+    bacteria_cell_age_in_current_timestep = [cs[it].cellAge for it in cs]
+
+    # growth rate
+    strainRate_rolling = [cs[it].strainRate_rolling for it in cs]
+    strainRate_rolling = [np.nan if x == 'nan' else x for x in strainRate_rolling]
+    bacteria_growth_rate_in_current_timestep = [element / dt for element in strainRate_rolling]
+
+    # convert to dataframe
+    bacteria_info = list(zip(bacteria_id_in_current_timestep,
+                             bacteria_label_in_current_timestep,
+                             bacteria_minor_in_current_timestep,
+                             bacteria_major_in_current_timestep,
+                             bacteria_first_end_points_x_in_current_timestep,
+                             bacteria_first_end_points_y_in_current_timestep,
+                             bacteria_second_end_points_x_in_current_timestep,
+                             bacteria_second_end_points_y_in_current_timestep,
+                             bacteria_orientation_in_current_time_step,
+                             bacteria_x_center_in_current_timestep,
+                             bacteria_y_center_in_current_timestep,
+                             bacteria_cell_age_in_current_timestep,
+                             bacteria_growth_rate_in_current_timestep))
+
+    columns_name = ['id', 'label', 'minor', 'major', 'x_end_point1', 'y_end_point1', 'x_end_point2', 'y_end_point2',
+                    'orientation', 'x_center', 'y_center', 'cell_age', 'growth_rate']
+    df_current_time_step = pd.DataFrame(bacteria_info, columns=columns_name)
+
+    return df_current_time_step
+
+
+def micro_colony_analysis(pickle_files_directory, summary_statistic_method, dt, min_size_of_micro_colony,
+                          max_distance_between_cells, um_pixel_ratio):
+    """
+    Goal: this is the main function; the function calls other function that are described above and
+    calculates aspect ratio for each micro colony in each time-step and store the outputs in png format and pickle.
+
+    @param summary_statistic_method   str     the method that we apply on the micro-colonies
+    
+    """
+
+    # store summary statistics of micro colonies
+    aspect_ratio_list = []
+    anisotropy_list = []
+    density_list = []
+    dist_vs_growth_rate_list = []
+
+    # read pickle files
+    path = pickle_files_directory + "/*.pickle"
+    filename_list = [filename for filename in sorted(glob.glob(path))]
+
+    # number of digits of last pickle file number
+    num_digit_last_pickle_file_num = len(str(len(filename_list)))
+
+    # In order to identify merged micro colonies, micro colony labels are stored.
```

Comment:
Identifying separate microcolonies should be optional to the user. This will waste resources when processing thousands of CellModeller simulations. Also, the experimental data may already be pre-processed to include only a single colony in a series of images, so finding merging of microcolonies would be extra overhead.

Reply:
all done! now the script is working on both user-preferred modes : local -global
unit test for a user convenience is provided.